File: backend-legacy/src/main/java/com/park/utmstack/domain/api_keys/ApiKey.java
package com.park.utmstack.domain.api_keys;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "api_keys")
public class ApiKey implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String apiKey;

    @Column
    private String allowedIp;

    @Column(nullable = false)
    private Instant createdAt;

    private Instant generatedAt;

    @Column
    private Instant expiresAt;
}

File: backend/modules/iam/domain/api_key.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type APIKey struct {
	ID          uuid.UUID  `gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	UserID      uuid.UUID  `gorm:"column:user_id;type:uuid;not null;index;uniqueIndex:ux_api_key_owner_name,priority:1"`
	TenantID    uuid.UUID  `gorm:"column:tenant_id;type:uuid;not null;index"`
	Name        string     `gorm:"size:255;not null;uniqueIndex:ux_api_key_owner_name,priority:2"`
	KeyHash     string     `gorm:"column:key_hash;size:64;not null;uniqueIndex"`
	AllowedIP   string     `gorm:"column:allowed_ip;type:text"`
	CreatedAt   time.Time  `gorm:"column:created_at;not null"`
	GeneratedAt time.Time  `gorm:"column:generated_at;not null"`
	ExpiresAt   *time.Time `gorm:"column:expires_at"`
}

func (APIKey) TableName() string { return "api_keys" }
